src/main/java/io/jenkins/pluginhealth/scoring/service/UpdateCenterService.java

```diff
@@ -50,16 +51,26 @@ public UpdateCenterService(ObjectMapper objectMapper, @Value("${jenkins.update.c
     public List<Plugin> readUpdateCenter() throws IOException {
         record UpdateCenterPlugin(String name, String scm, ZonedDateTime releaseTimestamp) {
             Plugin toPlugin() {
-                return new Plugin(this.name, this.scm, this.releaseTimestamp);
+                return new Plugin(this.name, this.scm, this.releaseTimestamp, new HashMap<String, String>());
             }
         }
-        record UpdateCenter(Map<String, UpdateCenterPlugin> plugins) {
+        record UpdateCenterDeprecations(String url) {
+        }
+        record UpdateCenter(Map<String, UpdateCenterPlugin> plugins, Map<String, UpdateCenterDeprecations> deprecations) {
         }
 
         UpdateCenter updateCenter = objectMapper.readValue(new URL(updateCenterURL), UpdateCenter.class);
-        return updateCenter.plugins.values().stream()
+        List<Plugin> pluginList = updateCenter.plugins.values().stream()
             .map(UpdateCenterPlugin::toPlugin)
             .collect(Collectors.toList());
+
+        for (Plugin plugin : pluginList) {
```

Comment:
I didn't feel happy to do this, because here I'm traversing the big 1800+ objects all over again to now add the "deprecation_reason" JSON object in them (wherever applicable). I understand this is now a scalable way to do this as it will slow things down as we move forward. Any thoughts on this?

Reply:
You are correct, you are iterating over the plugins again, which might not be as performant as it could be.
Maybe we could leverage the fact that we do that already once (in lines 60 to 62) and transform the plugins to hold this details before returning the list. Something like a map operation ;)